DirectProgramming/C++SYCL_FPGA/Tutorials/Features/task_sequences/task_sequence/src/main.cpp

```diff
@@ -0,0 +1,215 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/ext/intel/experimental/pipes.hpp>
+#include <sycl/ext/intel/experimental/task_sequence.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+constexpr int kVectSize = 128;
+
+// Forward declare the kernel names in the global scope.
+// This FPGA best practice reduces name mangling in the optimization reports.
+class IDOptimized;
+
+using ValueT = int;
+
+// Minimum capacity of a pipe.
+// Set to 0 to let compiler decides on the pipe capacity.
+constexpr size_t kPipeMinCapacity = 0;
+
+// Pipes 
+class PipeIn0_ID;
+using PipeIn0 = sycl::ext::intel::experimental::pipe<
+    // Usual pipe parameters
+    PipeIn0_ID,       // An identifier for the pipe
+    ValueT,           // The type of data in the pipe
+    kPipeMinCapacity  // The capacity of the pipe
+    >;
+
+class PipeIn1_ID;
+using PipeIn1 = sycl::ext::intel::experimental::pipe<
+    // Usual pipe parameters
+    PipeIn1_ID,       // An identifier for the pipe
+    ValueT,           // The type of data in the pipe
+    kPipeMinCapacity  // The capacity of the pipe
+    >;
+
+class PipeAB_ID;
+using PipeAB = sycl::ext::intel::pipe<
+    // Usual pipe parameters
+    PipeAB_ID,        // An identifier for the pipe
+    ValueT,           // The type of data in the pipe
+    kPipeMinCapacity  // The capacity of the pipe
+    >;
+
+class PipeBC_ID;
+using PipeBC = sycl::ext::intel::pipe<
+    // Usual pipe parameters
+    PipeBC_ID,        // An identifier for the pipe
+    ValueT,           // The type of data in the pipe
+    kPipeMinCapacity  // The capacity of the pipe
+    >;
+
+class PipeCD_ID;
+using PipeCD = sycl::ext::intel::pipe<
+    // Usual pipe parameters
+    PipeCD_ID,        // An identifier for the pipe
+    ValueT,           // The type of data in the pipe
+    kPipeMinCapacity  // The capacity of the pipe
+    >;
+
+class PipeAD_ID;
+using PipeAD = sycl::ext::intel::pipe<
+    // Usual pipe parameters
+    PipeAD_ID,        // An identifier for the pipe
+    ValueT,           // The type of data in the pipe
+    kPipeMinCapacity  // CThe capacity of the pipe
+    >;
+
+class PipeOut_ID;
+using PipeOut = sycl::ext::intel::experimental::pipe<
+    // Usual pipe parameters
+    PipeOut_ID,       // An identifier for the pipe
+    ValueT,           // The type of data in the pipe
+    kPipeMinCapacity  // The capacity of the pipe
+    >;
+
+[[intel::use_stall_enable_clusters]] 
+void loopA(int len) {
```

Comment:
```suggestion
void LoopA(int len) {
```
etc.

functions and types should be in `PascalCase`